File: admin/client/tools/requirementsCompliance.ts
import { z } from "zod";

export const requirementsComplianceToolConfig = z.object({
  name: z.literal("requirementsCompliance"),
  label: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
  enabled: z.boolean().nullable().optional(),
  config: z.object({}).nullable().optional(),
});
export type requirementsComplianceToolConfigType = z.infer<typeof requirementsComplianceToolConfig>;
export const DEFAULT_REQUIREMENTS_COMPLIANCE_TOOL_CONFIG = {
  label: "requirementsCompliance",
  description: "",
  config: {},
  enabled: false,
};

File: src/models/tools/requirements_compliance.py
from typing import ClassVar, Dict, Literal
from pydantic import BaseModel

class RequirementsComplianceTool(BaseModel):
    config_id: ClassVar[str] = "requirementsCompliance"
    name: Literal["requirementsCompliance"] = "requirementsCompliance"
    tool_type: Literal["local"] = "local"
    label: Literal["Requirements Compliance"] = "Requirements Compliance"
    description: str = "Generate a compliance report based on the requirements and description source."
    config: Dict = {}
    enabled: bool = False